src/coreclr/jit/copyprop.cpp

```diff
@@ -288,6 +273,71 @@ unsigned Compiler::optIsSsaLocal(GenTreeLclVarCommon* lclNode)
     return lclNum;
 }
 
+//------------------------------------------------------------------------------
+// optCopyPropPushDef: Push the new live SSA def on the stack for "lclNode".
+//
+// Arguments:
+//    asg        - The assignment node for this def (will be "nullptr" for "use" defs)
+//    lclNode    - The local tree representing "the def" (that can actually be a use)
+//    lclNum     - The local's number (see "optIsSsaLocal")
+//    curSsaName - The map of local numbers to stacks of their defs
+//
+void Compiler::optCopyPropPushDef(GenTreeOp*           asg,
+                                  GenTreeLclVarCommon* lclNode,
+                                  unsigned             lclNum,
+                                  LclNumToLiveDefsMap* curSsaName)
+{
+    assert((lclNum != BAD_VAR_NUM) && (lclNum == optIsSsaLocal(lclNode)));
+
+    // Shadowed parameters are special: they will (at most) have one use, that is one on the RHS of an
+    // assignment to their shadow, and we must not substitute them anywhere. So we'll not push any defs.
+    if ((gsShadowVarInfo != nullptr) && lvaGetDesc(lclNum)->lvIsParam &&
+        (gsShadowVarInfo[lclNum].shadowCopy != BAD_VAR_NUM))
+    {
+        return;
+    }
+
+    unsigned ssaDefNum = SsaConfig::RESERVED_SSA_NUM;
+    if (asg == nullptr)
+    {
+        // Parameters, this pointer etc.
+        assert((lclNode->gtFlags & GTF_VAR_DEF) == 0);
+        assert(lclNode->GetSsaNum() == SsaConfig::FIRST_SSA_NUM);
+        ssaDefNum = lclNode->GetSsaNum();
+    }
+    else
+    {
+        assert((lclNode->gtFlags & GTF_VAR_DEF) != 0);
+        // This will be "RESERVED_SSA_NUM" for promoted struct fields assigned using the parent struct.
+        // TODO-CQ: fix this.
+        ssaDefNum = GetSsaNumForLocalVarDef(lclNode);
```

Comment:
Assert here that `ssaDefNum != SsaConfig::FIRST_SSA_NUM` ?

Reply:
We will hit this path for first definitions of non-parameters, so I do not think we can do that.